chat.go

```diff
@@ -0,0 +1,75 @@
+package chat
+
+import (
+	"log"
+	"net"
+	"sync"
+)
+
+type Server struct {
+	sync.Mutex
+	listener net.Listener
+	running  bool
+}
+
+type Client struct {
+	connection net.Conn
+}
+
+func Start(addr string) (Server, error) {
```

Comment:
```suggestion
func Start(addr string) (*Server, error) {
```

It's always a question whether your constructor function should return a pointer, or a value. If most or all of your methods are pointer methods, which they are in this case, then returning a pointer makes sense:

```go
var s *chat.Server
s = chat.Start(":8080")
s.Stop()
```